File: src/main/java/com/conference/website/integration/MetricsClient.java
package com.conference.website.integration;

import reactor.core.publisher.Mono;

public interface MetricsClient {

    Mono<Long> incrementViews(Long talkId);

    Mono<Long> incrementLikes(Long talkId);

    Mono<Long> incrementAttends(Long talkId);

    Mono<Long> getViews(Long talkId);

    Mono<Long> getLikes(Long talkId);

    Mono<Long> getAttends(Long talkId);
}

File: src/main/kotlin/com/conference/website/integration/MetricsClient.kt
package com.conference.website.integration

import reactor.core.publisher.Mono

interface MetricsClient {

    fun incrementViews(talkId: Long): Mono<Long>

    fun incrementLikes(talkId: Long): Mono<Long>

    fun incrementAttends(talkId: Long): Mono<Long>

    fun getViews(talkId: Long): Mono<Long>

    fun getLikes(talkId: Long): Mono<Long>

    fun getAttends(talkId: Long): Mono<Long>
}
